specification/mzlb_json.schema.json

```diff
@@ -0,0 +1,141 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "HUPO-PSI mzlb spectral library specification",
+    "description": "Describe library, spectrum, and spectrum-component attributes",
+    "required": ["attributes", "format_version", "spectra"],
+    "type": "object",
+    "properties": {
+        "attributes": {
+            "type": "array",
+            "description": "A list of controlled vocabulary terms describing the spectral library",
+            "items": {
+                "$ref": "#/definitions/cvparam"
+            }
+        },
+        "spectra": {
+            "type": "array",
+            "description": "A list of spectra contained in the spectral library",
+            "items": {
+                "$ref": "#/definitions/spectrum"
+            }
+        },
+        "format_version": {
+            "type": "string",
+            "description": "The format specification's version name"
+        }
+    },
+    "definitions": {
+        "cvparam": {
+            "type": "object",
+            "description": "A controlled vocabulary term with an optional associated value, which may itself be a cvparam",
+            "required": ["accession", "name"],
+            "properties": {
+                "accession": {
+                    "type": "string",
+                    "description": "The accession number of the external controlled vocabulary term"
+                },
+                "name": {
+                    "type": "string",
+                    "description": "The human readable name of the controlled vocabular term"
+                },
+                "value": {
+                    "oneOf": [
+                        {"type": ["string", "number", "boolean", "null"]}
+                    ]
+                },
+                "value_accession": {
+                    "type": "string",
+                    "description": "The accession number of the external controlled vocabulary term describing the value of this cvparam"
+                },
+                "cv_param_group": {
+                    "type": "string",
+                    "description": "An identifier connecting a group of related parameters"
+                }
+            }
+        },
+        "analyte": {
+            "type": "object",
+            "description": "A molecule that is believed to have generated or contributed to the spectrum",
+            "required": ["attributes", "id"],
+            "properties": {
+                "attributes": {
+                    "type": "array",
+                    "description": "A list of controlled vocabulary terms describing the analyte",
+                    "items": {
+                        "$ref": "#/definitions/cvparam"
+                    }
+                },
+                "id": {
+                    "type": "string",
+                    "description": "A short identifier unique within the spectrum entry"
+                }
```

Comment:
This is linked my question on line 95. I thought that analyte IDs were only going to be integer numbers, starting with analyte 1. I guess the intention here is allow any string that is unique within the spectrum?

Does this work for the text serialization of the peak annotation format, where these IDs are referenced?

Reply:
The peak annotation parser regex region that is responsible for this is `(?P<analyte_reference>[^@\s]+)@)?`. This can tolerate anything that doesn't contain a @ or a space, so it can be any alphanumeric string. I think the spec does special case specific integers though, so I agree we can constrain this to integers.